wgpu-hal/src/metal/device.rs

```diff
@@ -584,13 +584,10 @@ impl crate::Device for super::Device {
             raw_cmd_buf: None,
             state: super::CommandState::default(),
             temp: super::Temp::default(),
+            counter: self.counters.command_encoders.clone(),
```

Comment:
Perhaps I'm missing something, but since there's no `Arc` here, this creates a counter that's entirely independent from the `Device`'s.

Reply:
Good catch, I saw `InternalCounter` is a wrapper around an `AtomicIsize` and noticed `InternalCounter` implements `Clone` (which I thought via a derive but that's not the case, moreover it couldn't have been a derive, `AtomicIsize` doesn't impl `Clone`, duh🙄).

Reply:
I wrapped `HalCounters` in an `Arc`.